tempesta_fw/sock.c

```diff
@@ -194,6 +194,9 @@ ss_do_send(struct sock *sk, SsSkbList *skb_list)
 	       smp_processor_id(), __func__,
 	       sk, tcp_send_head(sk), sk->sk_state);
 
+	if ((flags & SS_SEND_F_CONN_CLOSE) && tcp_close_state(sk))
+		ss_tcp_shutdown(sk);
```

Comment:
Perhaps I am missing something here, but I believe that the goal is not just set the FIN flag on the last skb of the outgoing message in the connection that is being closed, but also to **close** the connection right now, not on the next timer tick (SoftIRQ run). Yes, the connection is moved to a closing state. However, Tempesta's close procedure is somewhat different from the "lengthy" kernel's close procedure. Perhaps a discussion is needed here.